npm_and_yarn/lib/dependabot/npm_and_yarn/package_name.rb

```diff
@@ -11,6 +11,13 @@ class PackageName
           (?<name>[a-z0-9\-~][a-z0-9\-._~]*)         # capture package name
           \z                                         # end of string
       }xi.freeze                                     # multi-line/case-insensitive
+      SCOPED_TYPES_PACKAGE_REGEX = %r{
+          \A                                         # beginning of string
+          @#{DEFINITELY_TYPED_SCOPE}\/               # starts with @types/
+          (?<scope>.+)                               # capture scope
+          __                                         # scoped name separator
+          (?<name>.+)                                # capture name
+      }xi.freeze                                     # multi-line/case-insensitive
```

Comment:
Missing end of string anchor?

```suggestion
          \z                                         # end of string
      }xi.freeze                                     # multi-line/case-insensitive
```

Reply:
👍 All comments addressed together in [14bb063](https://github.com/dependabot/dependabot-core/pull/5025/commits/14bb0637d9a455e903ec498a4291c88c54011571)